main/src/ca/uwaterloo/flix/api/lsp/StackConsumer.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024 Alexander Dybdahl Troelsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp
+
+import ca.uwaterloo.flix.api.lsp.Visitor.Consumer
+import ca.uwaterloo.flix.language.ast.Ast.{AssocTypeConstructor, Derivation, Derivations, EqualityConstraint, TraitConstraint}
+import ca.uwaterloo.flix.language.ast.Type
+import ca.uwaterloo.flix.language.ast.TypedAst.Pattern.Record.RecordLabelPattern
+import ca.uwaterloo.flix.language.ast.TypedAst.*
+import ca.uwaterloo.flix.language.ast.shared.Annotation
+import ca.uwaterloo.flix.language.ast.shared.SymUse.*
+
+/**
+  * Consumer that collects every visited AST nodes on a stack where the head is the last element visited.
```

Comment:
A bit more information about what is actually computed.

Reply:
yep